src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -157,6 +158,45 @@
      */
     List<SimpleInterval> intervalsForTraversal;
 
+    /**
+     * Get the {@link ReferenceDataSource} for this {@link GATKTool}.
+     * Will throw a {@link GATKException} if the reference is null.
+     * Clients are expected to call the {@link #hasReference()} method prior to calling this.
+     * @return the {@link ReferenceDataSource} for this {@link GATKTool}.  Never {@code null}.
+     */
+    protected ReferenceDataSource getReferenceDataSource() {
```

Comment:
Rename to `directlyAccessEngineReferenceDataSource()` to deter casual callers and improve the odds that they'd check the javadoc.

Reply:
Fixed.